### PR TITLE
feat(behavior_path_planner): expand hatched road markings

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

Currently, we cannot utilize the area of hatched road markings for obstacle avoidance.

In the following discussion, the area of hatched road markings is defined, and with this PR we can expand the drivable area with the area of hatched road markings.
https://github.com/orgs/autowarefoundation/discussions/3414

launcher PR: https://github.com/autowarefoundation/autoware_launch/pull/337

### code change
- add `enable_expanding_hatched_road_markings` in DrivableAreaInfo
    - In avoidance module, this variable may be enabled depending on `use_hatched_road_markings` ros parameters.
    - This ros parameter is set to false by default since in Turkey it is prohibited to drive in the area of hatched road markings.
- implement the feature of expanding the drivable area with hatched road markings polygons in `generateDrivableArea` function.
    - For old architecture, the argument of `enable_expanding_hatched_road_markings` is set to false.

### result
exampl1 
image
In the above figure, the yellow polygon is the hatched road markings.
image
image

example2
image
In the above figure, the yellow polygon is the hatched road markings.
image
image

## Tests performed

planning simulator

## Effects on system behavior

Can use hatched road markings for obstacle avoidance

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by purewater0901 — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/src/scene_module/avoidance_by_lc/module.cpp [hunk 1]
```diff
@@ -0,0 +1,617 @@
+// Copyright 2023 TIER IV, Inc.
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/src/scene_module/avoidance_by_lc/module.cpp:1]: This file is no longer needed. Removed by https://github.com/autowarefoundation/autoware.universe/pull/3632.

**Review by satoshi-ota — CHANGES_REQUESTED** [review]
Could you remove avoidance_by_lc directory?

It is removed by https://github.com/autowarefoundation/autoware.universe/pull/3632.

File: planning/behavior_path_planner/src/scene_module/avoidance_by_lc/module.cpp (hunk 1 above)
Comment (takayuki5168) [reply planning/behavior_path_planner/src/scene_module/avoidance_by_lc/module.cpp]: Thank you.
Done. https://github.com/autowarefoundation/autoware.universe/pull/3604/commits/b23f68ee65f42c05837a96effc3646839d6a5c79

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**